src/Lecture3.hs

```diff
@@ -122,10 +139,10 @@ data Reward = Reward
     } deriving (Show, Eq)
 
 instance Semigroup Reward where
-
+  (<>) (Reward rg1 rs1) (Reward rg2 rs2) = Reward ((<>) rg1 rg2) (rs1 || rs2)
 
 instance Monoid Reward where
-
+  mempty = Reward (Gold 0) False
```

Comment:
You can use `mempty` for `Gold` here because it already has a `Monoid` instance 🏅 
```suggestion
  mempty = Reward mempty False
```